CONTRIBUTING.md

```diff
@@ -36,3 +36,12 @@ To set up your development environment:
 1. `make build`: recompile your code after modifying any Rust code in `src/`
 2. `make test`: run tests
 3. `DAFT_RUNNER=ray make test`: set the runner to the Ray runner and run tests (DAFT_RUNNER defaults to `py`)
+
+### Developing with Ray
+
+The simplest way to run a development version of Daft on Ray is to build on the same operating system version as your Ray nodes, in order to ensure that your binaries are executable on Ray.
```

Comment:
We should clarify that this is for running a remote Ray cluster?

Local Ray cluster doesn't need these setup steps I think